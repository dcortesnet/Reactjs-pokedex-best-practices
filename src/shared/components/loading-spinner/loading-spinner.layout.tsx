import React from "react";
import "./loading-spinner.css";

export const LoadingSpinnerLayout = () => {
  return (
    <div className="loader__container">
      <div className="loader"></div>
    </div>
  );
};
