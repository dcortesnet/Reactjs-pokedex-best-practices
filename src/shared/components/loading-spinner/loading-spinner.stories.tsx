import { Story, Meta } from "@storybook/react";
import { LoadingSpinnerLayout } from "./loading-spinner.layout";

export default {
  title: "Shared/components/LoadingSpinner",
  component: LoadingSpinnerLayout,
} as Meta;

export const Template: Story<any> = (args) => <LoadingSpinnerLayout />;
