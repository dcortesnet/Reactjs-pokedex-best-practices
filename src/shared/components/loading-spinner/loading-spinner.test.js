import { LoadingSpinnerLayout } from "./loading-spinner.layout";
import React from "react";
import Enzyme, { shallow, mount } from "enzyme";
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import toJson from "enzyme-to-json";

Enzyme.configure({ adapter: new Adapter() });

describe("LoadingSpinnerLayout", () => {
  it("Renders correctly", () => {
    const r = shallow(<LoadingSpinnerLayout></LoadingSpinnerLayout>);

    expect(toJson(r)).toMatchSnapshot();
  })
});
