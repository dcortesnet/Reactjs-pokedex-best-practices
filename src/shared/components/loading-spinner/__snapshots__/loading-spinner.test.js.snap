// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoadingSpinnerLayout Renders correctly 1`] = `
<div
  className="loader__container"
>
  <div
    className="loader"
  />
</div>
`;
